packages/@react-aria/tabs/src/useTabs.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {AriaLabelingProps, CollectionBase, Node, Orientation, SingleSelection} from '@react-types/shared';
+import {HTMLAttributes, Key, RefObject, useMemo} from 'react';
+import {ListState} from '@react-stately/list';
+import {TabsKeyboardDelegate} from '.';
+import {useLocale} from '@react-aria/i18n';
+import {usePress} from '@react-aria/interactions';
+import {useSelectableCollection, useSelectableItem} from '@react-aria/selection';
+
+interface TabsProps<T> extends CollectionBase<T>, SingleSelection, AriaLabelingProps {
```

Comment:
Could these types be moved into `@react-types/tabs`? Then the Spectrum props could extend from this type.